src/main/scala/io/github/pauljamescleary/petstore/config/DatabaseConfig.scala

```diff
@@ -18,23 +19,12 @@ object DatabaseConfig {
 
   /**
     * Runs the flyway migrations against the target database
-    *
-    * This only gets applied if the database is H2, our local in-memory database.  Otherwise
-    * we skip this step
     */
-  def initializeDb[F[_]](dbConfig: DatabaseConfig)(implicit S: Sync[F]): F[Unit] =
-    if (dbConfig.url.contains(":h2:")) {
-      S.delay {
-        val fw = new Flyway()
-        val ds = new org.h2.jdbcx.JdbcDataSource()
-        ds.setUrl(dbConfig.url)
-        ds.setUser(dbConfig.user)
-        ds.setPassword(dbConfig.password)
-        fw.setDataSource(ds)
-        fw.migrate()
-        ()
-      }
-    } else {
-      S.pure(())
+  def initializeDb[F[_]](ds: DataSource)(implicit S: Sync[F]): F[Unit] =
+    S.delay {
+      val fw = new Flyway()
+      fw.setDataSource(ds)
+      fw.migrate()
```

Comment:
ok, looks like we can `fw.setDataSource(url, user, pass)` instead of passing the DataSource.

We can avoid hard-coding to h2 if we pass in the `DatabaseConfig`

Reply:
That does work actually, good catch. Even after reading the flyway source I'm not sure how it finds the correct driver based on the URL, but it does seem to work well.